hazelcast-client/src/main/java/com/hazelcast/client/impl/HazelcastClientInstanceImpl.java

```diff
@@ -299,18 +306,80 @@ private MetricsRegistryImpl initMetricsRegistry() {
             addressProviders.add(new DiscoveryAddressProvider(discoveryService, loggingService));
         }
 
+        ClientAwsConfig awsConfig = networkConfig.getAwsConfig();
+        AwsAddressProvider awsAddressProvider = initAwsAddressProvider(awsConfig);
+        if (awsAddressProvider != null) {
+            addressProviders.add(awsAddressProvider);
+        }
+
+        HazelcastCloudAddressProvider cloudAddressProvider = initCloudAddressProvider();
+        if (cloudAddressProvider != null) {
+            addressProviders.add(cloudAddressProvider);
+        }
+
+        addressProviders.add(new DefaultAddressProvider(networkConfig, addressProviders.isEmpty()));
+        return addressProviders;
+    }
+
+    private HazelcastCloudAddressProvider initCloudAddressProvider() {
+        String cloudToken = properties.getString(ClientProperty.HAZELCAST_CLOUD_TOKEN);
+        if (cloudToken != null) {
```

Comment:
should we also do check for against an empty string as well?

Reply:
I think empty string is same as an invalid token. No need the put an extra check for that.